docs/build/guides/conventions/soroban-contract-init-template.mdx

```diff
@@ -0,0 +1,339 @@
+
+```bash
+npm install dotenv glob --dev
+```
+
+### Writing the initialization code
+
+As a general overview, the following script will perform the below tasks:
+
+1. Create network profiles for contract project
+2. Generate and fund account for interaction with the Stellar network
+3. Build smart contract .wasm files
+4. Deploy smart contract to the Stellar network
+5. Generate smart contract Typescript bindings
+
+#### Importing the modules
+
+The following modules allow us to work with environment variables, execute Stellar CLI commands, search files and directories, work with paths, and create and delete files, in that order.
+
+```javascript
+import "dotenv/config";
+import { execSync } from "child_process";
+import { globSync } from "glob";
+import path from "path";
+import { writeFileSync, rmSync } from "fs";
+```
+
+### Creating the wrapper function
+
+Instead of calling the `execSync` function and making the error handling process multiple times, we create a wrapper function to encapsulate this function.
+
+Furthermore, when deploying the contract, the command returns the deployed contract ID, which is required when generating contract Typescript bindings. Wrapping the `execSync` function will enable to return Stellar CLI command output while handling errors in one go.
+
+```javascript
+const run = (command) => {
+  try {
+    const output = execSync(command, { stdio: ["inherit", "pipe", "pipe"] });
+    console.log("Here comes the output\n");
+    return output.toString();
+  } catch (error) {
+    console.error("Error executing command: \n", error.message);
+    return null; // or throw error if you want to propagate it
+  }
+};
+```
+
+### Creating network profiles
+
+This code snippet executes a Stellar CLI command to add a network profile. It uses environment variables (process.env.NETWORK, process.env.RPC_URL, process.env.NETWORK_PASSPHRASE) to specify the network details and RPC URL.
+
+You can clearly see the benefit of wrapping the `execSync` function as it results in a much cleaner code and short snippet while retaining same functionality.
+
+```javascript
+run(
+  `stellar network add ${process.env.NETWORK} --rpc-url ${process.env.RPC_URL} --network-passphrase ${process.env.NETWORK_PASSPHRASE}`,
+);
+```
+
+### Creating account profiles
+
+This code generates Stellar key pairs for an account (process.env.ACCOUNT) and funds them accordingly on a specific network (process.env.NETWORK).
+
+```javascript
+run(
+  `stellar keys generate ${process.env.ACCOUNT} --network ${process.env.NETWORK}`,
+);
+```
+
+### Building the contract
+
+This snippet builds Stellar smart contracts in the project directory.
+
+```javascript
+run("stellar contract build");
+```
+
+### Deploying the contract
+
+This code deploys one or more Stellar smart contracts. It uses files found in target/wasm32-unknown-unknown/release/\*.wasm, deploys each with specified parameters, and stores contract details (alias and contractId) in contractsObject.
```

Comment:
```suggestion
This code deploys one or more Stellar smart contracts. It uses files found in `target/wasm32-unknown-unknown/release/*.wasm`, deploys each with specified parameters, and stores contract details (alias and contractId) in contractsObject.
```

When writing filepaths or filenames (outside of scripts), let's use backticks to separate them and make it a bit easier for the eye to catch what's happening.